app/controllers/api/v1/task_category_controller.rb

```diff
@@ -0,0 +1,53 @@
+class Api::V1::TaskCategoryController < ApplicationController
+  before_action :require_admin, only: [:delete, :update, :create]
+
+  def index 
+    service = TaskCategoryService.new
+    list_all = service.find_all
+    if service.errors.present?
+      render json: { success: false, error: service.errors }, status: :unprocessable_entity
+    else
+      render json: { success: true, data: list_all.map { |category| TaskCategorySerializer.new(category) } }, status: :ok
+    end
+  end
+
+  def delete
+    service = TaskCategoryService.new 
+    service.delete_all(params[:name])
+    if service.errors.present?
+      render json: { success: false, error: service.errors }, status: :unprocessable_entity
+    else
+      render json: { success: true, message: 'Task Category Successfully Deleted' }, status: :created
+    end
+  end
+
+  def update
+    service = TaskCategoryService.new
+    service.update(params[:id], category_params)
+    if service.errors.present?
+      render json: { success: false, error: service.errors }, status: :unprocessable_entity
+    else
+      render json: { success: true, message: 'Task Category Successfully Updated' }, status: :created
```

Comment:
return serialised task category from here rather than a message as I asked earlier. this is being repeated in many places, please always make sure you are address a PR throughout the PR.